samza-api/src/main/java/org/apache/samza/operators/KV.java

```diff
@@ -45,4 +49,30 @@ public K getKey() {
   public V getValue() {
     return value;
   }
+
+  @Override
+  public boolean equals(Object other) {
+    if (this == other) {
+      return true;
+    }
+    if (!(other instanceof KV)) {
+      return false;
+    }
+    KV<?, ?> otherKv = (KV<?, ?>) other;
+    return Objects.deepEquals(this.key, otherKv.key)
```

Comment:
Please add a comment about why we need `deepEquals` and that this implementation comes from Beam.

Reply:
Added.